src/modules/map/components/LocationPopup.vue

```diff
@@ -126,101 +135,140 @@ export default {
-            this.generateQrCodeFromBackend()
+        '$route.query'(newQuery) {
+            if (this.clickInfo && this.displayLocationPopup) {
+                this.updateShareLink(this.clickInfo.coordinate, newQuery)
+            }
         },
     },
     methods: {
         ...mapActions(['clearClick']),
         onClose() {
             this.clearClick()
+            this.qrCodeImageSrc = null
+            this.what3Words = ''
+            this.height = null
+            this.shareLinkUrlShorten = null
+            this.shareLinkUrl = null
         },
-        reprojectClickCoordinates(targetEpsg) {
-            return proj4(CoordinateSystems.WEBMERCATOR.epsg, targetEpsg, this.clickCoordinates)
-        },
-        requestWhat3WordBackend() {
-            if (this.displayLocationPopup) {
-                registerWhat3WordsLocation(this.clickCoordinates, this.currentLang).then(
-                    (what3word) => {
-                        this.clickWhat3Words = what3word
-                    }
-                )
-            }
+        updateWhat3Word(coordinate, lang) {
+            registerWhat3WordsLocation(coordinate, lang)
+                .then((what3word) => {
+                    this.what3Words = what3word
+                })
+                .catch(() => {
+                    log.error(`Failed to retrieve What3Words Location`)
+                    this.what3Words = ''
+                })
         },
-        registerHeigthFromBackend() {
-            if (this.displayLocationPopup) {
-                requestHeight(this.clickCoordinates).then((height) => {
+        updateHeight(coordinate) {
+            requestHeight(coordinate)
+                .then((height) => {
                     this.height = height
                 })
-            }
+                .catch(() => {
+                    log.error(`Failed to get position height`)
+                    this.height = null
+                })
         },
-        async generateQrCodeFromBackend() {
-            if (this.displayLocationPopup) {
-                this.qrCodeImageSrc = await generateQrCode(this.shareLinkUrl)
+        updateShareLink(coordinate, routeQuery) {
+            let [lon, lat] = reproject(CoordinateSystems.WGS84.epsg, coordinate)
+            let query = {
+                ...routeQuery,
+                crosshair: 'marker',
+                lat,
+                lon,
             }
+            this.shareLinkUrl = `${location.origin}/#/map?${stringifyQuery(query)}`
+            this.shortenShareLink(this.shareLinkUrl)
+        },
+        shortenShareLink(url) {
+            createShortLink(url)
+                .then((shortUrl) => {
+                    this.shareLinkUrlShorten = shortUrl
+                    this.updateQrCode(shortUrl)
+                })
+                .catch((err) => {
+                    log.error(`Failed to shorten Share URL`)
+                    this.shareLinkUrlShorten = null
+                })
+        },
```

Comment:
as you are calling two promise based function in a row (or nested), it could enhance readability to switch to an `async...await` call here, and wrap the whole thing in `try...catch` block

Reply:
Done